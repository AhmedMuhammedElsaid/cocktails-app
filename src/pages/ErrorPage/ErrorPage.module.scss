@import "../../scss/variables.scss";
.ErrorPage {
  min-height: 80vh;
  &__status {
    margin: 0 0 40px;
    font-size: 4em;
    line-height: 1;
    letter-spacing: 0.15em;
  }
  &__message {
    font-size: 2em;
    font-weight: 500;
    text-transform: capitalize;
  }

  &__BtnHolder {
    margin-block-start: 2em;
  }
  &__Btn {
    padding: .5em;
    border-radius: 0.7em;
    background: $primaryColor;
    color: $mainWhite;
    font-size: 1.3em;
    margin-block-start: 2em;
    text-transform: capitalize;

    text-decoration: none;
  }
}
