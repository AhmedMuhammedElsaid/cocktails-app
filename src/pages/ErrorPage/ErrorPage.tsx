import React, { FunctionComponent } from "react";
import { Grid } from "@material-ui/core";
import { Fade } from "react-reveal";
import styles from "./ErrorPage.module.scss";
import { Link } from "react-router-dom";

const ErrorPage: FunctionComponent = () => {
  return (
    <Fade top duration={2000} distance="40px">
      <Grid
        className={styles.ErrorPage}
        container
        direction="column"
        justifyContent="center"
        alignItems="center"
      >
        <h1 className={styles.ErrorPage__message} data-testid="error-msg-testid">
          Ooops!!,Something Went Wrong
        </h1>
        <span className={styles.ErrorPage__BtnHolder} data-testid="go-home-testid">
          <Link rel="preconnect" className={styles.ErrorPage__Btn} to="/">
            Go back to home
          </Link>
        </span>
      </Grid>
    </Fade>
  );
};

export default ErrorPage;
