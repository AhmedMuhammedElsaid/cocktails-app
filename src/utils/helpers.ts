/*
 @params object
 return object with no null ,undefined , empty string "cleaned object"
*/
export function cleanObject(object: { [x: string]: any }) {
  if (!object) return {};
  if (typeof object === "object") {
    for (let propName in object) {
      if (
        object[propName] === null ||
        object[propName] === undefined ||
        object[propName]?.length === 0
      ) {
        delete object[propName];
      }
    }
  }
  return object;
}
