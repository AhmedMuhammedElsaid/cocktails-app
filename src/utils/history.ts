import { createBrowserHistory } from "history";

const history = createBrowserHistory();
// method to redirect the user to error page in case any error happens while fetching the data from the server
export const redirectToErrorPage = () => {
  history.push(`/pagenotfound`);
  history.go(0);
};
export default history;
