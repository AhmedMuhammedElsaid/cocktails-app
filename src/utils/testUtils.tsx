import { render, RenderResult } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

// this is a handy function that I would utilize for any component
// that relies on the router being in context
export const customRender = (component: JSX.Element): RenderResult => {
  return {
    ...render(<Router>{component}</Router>),
  };
};
