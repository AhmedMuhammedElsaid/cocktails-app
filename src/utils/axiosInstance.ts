import axios from "axios";
import { redirectToErrorPage } from './history';
import { toast } from "react-toastify";

const axiosInstance = axios.create({
  // setting the default uri for this instance
  baseURL: process.env.REACT_APP_BASE_URI,
});

// setting language  configuration on every single request we send to the BE
axiosInstance.interceptors.request.use(
  (request: any) => {
    return {
      ...request,
      headers: {
        ...request.headers,
        // setting the header lang german "de"
        "Accept-Language": "de",
      },
    };
  },
  (error) => {
    return Promise.reject(error);
  }
);

axiosInstance.interceptors.response.use(
  (response) => {
    return response.data;
  },
  (error) => {
    switch (error?.response?.status) {
      case 400:
        return {
          data: null,
          hasError: true,
          error: [error.response.data],
        };
      case 401:
        toast.error("Unauthorized Call Access");
        break;
      case 404:
        toast.error("404 Not Found");
        break;
      case 500:
        toast.error("Server Is Down");
        break;
    }
    redirectToErrorPage()
    return Promise.reject(error);
  }
);
export default axiosInstance;
