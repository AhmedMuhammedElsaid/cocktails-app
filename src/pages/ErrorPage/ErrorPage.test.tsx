import { customRender } from "utils/testUtils";
import ErrorPage from "./ErrorPage";

const messageHolder = "error-msg-testid",
  homeBtn = "go-home-testid";
describe("Test error page rendering without crashing", () => {
  it("renders without error", () => {
    const { getByTestId, asFragment } = customRender(<ErrorPage />);
    expect(getByTestId(messageHolder)).toBeInTheDocument();
    expect(getByTestId(homeBtn)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
