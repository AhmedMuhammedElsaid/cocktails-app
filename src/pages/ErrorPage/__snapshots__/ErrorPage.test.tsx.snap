// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test error page rendering without crashing renders without error 1`] = `
<DocumentFragment>
  <div
    class="react-reveal"
  >
    <div
      class="MuiGrid-root ErrorPage MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
    >
      <h1
        class="ErrorPage__message"
        data-testid="error-msg-testid"
      >
        Ooops!!,Something Went Wrong
      </h1>
      <span
        class="ErrorPage__BtnHolder"
        data-testid="go-home-testid"
      >
        <a
          class="ErrorPage__Btn"
          href="/"
          rel="preconnect"
        >
          Go back to home
        </a>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
